docs/deploy/bnns.rst

```diff
@@ -0,0 +1,172 @@
+..  Licensed to the Apache Software Foundation (ASF) under one
+    or more contributor license agreements.  See the NOTICE file
+    distributed with this work for additional information
+    regarding copyright ownership.  The ASF licenses this file
+    to you under the Apache License, Version 2.0 (the
+    "License"); you may not use this file except in compliance
+    with the License.  You may obtain a copy of the License at
+
+..    http://www.apache.org/licenses/LICENSE-2.0
+
+..  Unless required by applicable law or agreed to in writing,
+    software distributed under the License is distributed on an
+    "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+    KIND, either express or implied.  See the License for the
+    specific language governing permissions and limitations
+    under the License.
+
+Relay BNNS Integration
+==========================
+**Author**: `Egor Churaev <https://github.com/echuraev>`_
+
+Introduction
+------------
+
+Apple BNNS library is a collection of functions that can be used to construct neural networks
+for inference (and train). It’s supported in macOS, iOS, tvOS, and watchOS. BNNS provides
+primitives executed on all CPU supported on those platforms and optimized for high performance
+and low-energy consumption. This integration will offload as many operators as possible from Relay to BNNS.
```

Comment:
Also should be noted:
* BNNS is a part of platform API and available on all modern apple systems. Does't required any external dependencies.

Possible benefits:
* Allow to use Apple private HW optimisation which doesn't accessible outside. Like AMX apple cpu extension.  
* Reducing binary size. No need to keep TVM compiled kernels, just transfer it into system BNNS call.   


> This integration will offload as many operators as possible
It's better to formulate "This integration allow to transfer operation execution from Relay to BNNS." 
Greedy BNNS notation is an option, not mandatory pass.

Reply:
Addressed. slightly modified - removed mentioning of binary size. It is questionable statement. TVM code gen will produce much less size than it will be in BNS runtime.